### PR TITLE
feat: Add pylhe recipe

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves https://github.com/scikit-hep/pylhe/issues/128

Recipe generated with [`grayskull` `v1.3.1`](https://github.com/conda-incubator/grayskull/releases/tag/v1.3.1).

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
> GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

@lukasheinrich @eduardo-rodrigues Can you please comment here that you confirm that you're willing to be a co-mainainter of the recipe with me

File: recipes/pylhe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - awkward >=1.2.0
+    - python-graphviz >=0.12.0
+    - networkx >=2.2
+    - particle >=0.16
+    - python >=3.6
+    - tex2pix >=0.3
```
Comment (matthewfeickert) [inline recipes/pylhe/meta.yaml:27]: `tex2pix` isn't on Conda-forge and I guess this is problematic, so I'll need to figure out how to tell Conda-forge about the sdist on https://pypi.org/project/tex2pix/.

File: recipes/pylhe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
```
Comment (matthewfeickert) [inline recipes/pylhe/meta.yaml:20]: Should also add `graphviz` dependency.

c.f. https://github.com/scikit-hep/pylhe/blob/c0d6c1b54039eac715fca5b18696fc82619c2da9/.github/workflows/ci.yml#L42

**Comment by eduardo-rodrigues** [comment]
> > GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
> 
> @lukasheinrich @eduardo-rodrigues Can you please comment here that you confirm that you're willing to be a co-mainainter of the recipe with me?

OK :).

File: recipes/pylhe/meta.yaml (hunk 2 above)
Comment (eduardo-rodrigues) [reply recipes/pylhe/meta.yaml:20]: Am going to be offline. Very quickly - for meta.yml you need `python-graphviz`, I believe, as I have in DecayLanguage.

File: recipes/pylhe/meta.yaml (hunk 1 above)
Comment (matthewfeickert) [reply recipes/pylhe/meta.yaml:27]: https://conda-forge.org/docs/maintainer/adding_pkgs.html#avoid-external-dependencies:

> As a general rule: all dependencies have to be packaged by conda-forge as well. This is necessary to assure [ABI](https://conda-forge.org/docs/misc/00_intro.html#term-ABI) compatibility for all our packages.

So either need to package `tex2pix` or address https://github.com/scikit-hep/pylhe/issues/53.

Comment (eduardo-rodrigues) [reply recipes/pylhe/meta.yaml:27]: Personally I would rather deal with https://github.com/scikit-hep/pylhe/issues/53. The making of a PDF from a figure is indeed something very decoupled from `pylhe`'s goals and scope, and we should avoid unnecessary (package) couplings. The production of a figure is something easy following https://github.com/scikit-hep/pylhe/pull/118, anyway, and it is now more a matter of showing an example of how to produce a PNG or PDF, see for example https://graphviz.readthedocs.io/en/stable/api.html?highlight=pdf#id3.

I can probably help on these matters in the next few days ... Let me know.

Comment (eduardo-rodrigues) [reply recipes/pylhe/meta.yaml:27]: See https://github.com/scikit-hep/pylhe/pull/131.

Comment (eduardo-rodrigues) [reply recipes/pylhe/meta.yaml:27]: MR https://github.com/scikit-hep/pylhe/pull/131 is now merged. I will remove the 2 obsoleted dependencies here. Of course the finalisation of this MR will require a release of `pylhe`, to be done soon.

File: recipes/pylhe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  run:
+    - awkward >=1.2.0
+    - python-graphviz >=0.12.0
+    - networkx >=2.2
```
Comment (eduardo-rodrigues) [inline recipes/pylhe/meta.yaml:24]:
```suggestion
```

File: recipes/pylhe/meta.yaml (hunk 1 above)
Comment (eduardo-rodrigues) [inline recipes/pylhe/meta.yaml:27]:
```suggestion
```

**Comment by eduardo-rodrigues** [comment]
Hello @matthewfeickert, let me know if you need help here now that you released pylhe 0.4.0. In any case I will add pylhe to scikit-hep once this is ticked off :-).

**Comment by matthewfeickert** [comment]
@conda-grayskull show requirements

**Comment by conda-grayskull** [comment]
Working on your request...

**Comment by matthewfeickert** [comment]
> I am not sure if this bot will work, I still need to fix a few things in it, sorry :/

That's fine! I was more trying it out just to see if it could pick up things in a staged-recipes request or only on a feedstock given https://twitter.com/mdtrevisani/status/1487894867799859201.

**Comment by matthewfeickert** [comment]
@chrisburr this should be ready for review.

**Pull request merged**